cli/src/lib/proxy.test.js

```diff
@@ -0,0 +1,45 @@
+const { rewriteLocation } = require('./proxy')
+
+describe('rewriteLocation', () => {
+    it('rewrites locations if they match the proxy target', () => {
+        const baseUrl = 'http://localhost:8080'
+
+        expect(
+            rewriteLocation({
+                location: 'https://play.dhis2.org/dev/login.action',
+                target: 'https://play.dhis2.org/dev',
+                baseUrl,
+            })
+        ).toBe(`${baseUrl}/login.action`)
+
+        expect(
+            rewriteLocation({
+                location: 'https://play.dhis2.org/dev/page?param=value',
+                target: 'https://play.dhis2.org/dev',
+                baseUrl,
+            })
+        ).toBe(`${baseUrl}/page?param=value`)
```

Comment:
Are there any other edge cases that should be tested here?

Reply:
Maybe cases where the target has a non-80 port?  And/or if the scheme is different (http vs https)

Reply:
Also where the target pathname occurs twice in the location pathname, as mentioned above